### ISSUE TITLE
3.0.0-alpha.2, format cannot be used as value

### ISSUE THREAD

**Issue by Nikola-Milovic** [date-fns/date-fns#3588]
I installed `date-fns` and it installed the `alpha` version by default, I haven't seen this reported before so here it is

```typescript
import { format } from 'date-fns';

const createdAt = project.createdAt ? format(project.createdAt.toDate(), 'PP') : 'N/A';
```

Which gives me the following error `'format' cannot be used as a value because it was exported using 'export type'.`
The format is exported as `export declare function format<DateType extends Date>(date: DateType | number | string, formatStr: string, options?: FormatOptions): string;`

` typescript 5.3.2` with `date-fns 3.0.0-alpha.2`

**Comment by graup** [comment]
Same on 3.0.0-rc2 when importing like this: `import { subDays } from "date-fns";` in Next.js 14.0.3, TS 5.3.2.

```
src/app/Container.tsx:45:7 - error TS1362: 'subDays' cannot be used as a value because it was exported using 'export type'.

45       subDays(new Date(), 7),
         ~~~~~~~

  node_modules/date-fns/index.d.mts:1:1
    1 export type * from "./index.d.ts";
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'subDays' was exported here.
 ```

**Comment by FedericoBiccheddu** [comment]
Same problem with `date-fns@3.0.0-rc.2` and `typescript`@`5.3.3`.

It resolves correctly if using:
```ts
import * as DateFNS from "date-fns";

DateFNS.formatISO(...)
```

**Comment by Nikola-Milovic** [comment]
Yeah but you lose tree-shaking, so its only a temporary solution :/ @FedericoBiccheddu

**Comment by UkonnRa** [comment]
Same problem, any update?

**Comment by cj-christoph-gysin** [comment]
Same problem on the release 3.0.0

**Comment by wojtekmaj** [comment]
If you change

```ts
export type * from "./index.d.ts";
```

to 

```ts
export * from "./index.d.ts";
```

in `index.d.mts` the error goes away 🤔

**Comment by b-maslennikov** [comment]
@wojtekmaj confirming. waiting for the fix

**Comment by claudfuen** [comment]
+1 waiting for fix

**Comment by hugotox** [comment]
Wow I got really exited about today's 3.0 release but this is still broken.

For now you can import the default and use the member function:

```typescript
  import DateFns from 'date-fns/format'

  DateFns.format('...')
```

**Comment by MattTrilsbach** [comment]
Having the same issue with the `format` function. I can confirm the fix suggested by @wojtekmaj works in the interim:

> If you change
> 
> ```ts
> export type * from "./index.d.ts";
> ```
> 
> to
> 
> ```ts
> export * from "./index.d.ts";
> ```
> 
> in `index.d.mts` the error goes away 🤔
> 
> Similar of course would need to be done to all entry files.

**Comment by nstuyvesant** [comment]
The fix also adds an error...
> A declaration file cannot be imported without 'import type'. Did you mean to import an implementation file './index.ts' instead?ts(2846)

**Comment by kossnocorp** [comment]
Can someone please produce a minimal reproduction repo? It will help to fix the problem.

We have a set of different environments [where it's all good](https://github.com/date-fns/date-fns/actions/runs/7247215749/job/19740705115): https://github.com/date-fns/date-fns/tree/main/examples

Sorry that it slipped!

**Comment by Sailwayfun** [comment]
https://github.com/Sailwayfun/date-fns-format.git

This is a newly set up React + Vite + Typescript project, where you can see Typescript complains about the "format" function.

**Comment by kossnocorp** [comment]
@Sailwayfun I appreciate it a lot! Can you please check if `date-fns@3.0.1-rc.1` helps it in your project?

**Comment by Sailwayfun** [comment]
@kossnocorp  I tried `date-fns@3.0.1-rc.1` with `format` and `addDays` in the same app, seems to work fine. By the way, can I still PR the example to the repo you mentioned in #3618?

**Comment by kossnocorp** [comment]
@Sailwayfun thank you, but I already included and adjusted your code as the example. Appreciate the prompt response! `3.0.1` is on the way!

**Comment by kossnocorp** [comment]
I shipped with fix with `date-fns@3.0.1`, please try it

**Comment by Sailwayfun** [comment]
> I shipped with the fix with `date-fns@3.0.1`, please try it

I updated the package and it worked fine. Thanks a lot!

**Comment by FedericoBiccheddu** [comment]
> I shipped with the fix with `date-fns@3.0.1`, please try it

Thank you for the fix and your work. Very much appreciated :pray:
